### PR TITLE
hushboard: init at 2021-03-17

### PR DESCRIPTION
**Pull request by keysmashes**
###### Motivation for this change
[Hushboard](https://kryogenix.org/code/hushboard/) mutes your microphone automatically while typing.

This is the first derivation I'm contributing to Nixpkgs, it could undoubtedly be better.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/hushboard/default.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+, librsvg
+, pycairo
+, pygobject3
+, six
+, xlib
+}:
+
+buildPythonApplication {
+  pname = "hushboard";
+  version = "2021-03-17";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/hushboard/default.nix:18]:
```suggestion
  version = "unstable-2021-03-17";
```

File: pkgs/applications/audio/hushboard/default.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  propagatedBuildInputs = [
+    pycairo
+    pygobject3
+    six
+    xlib
+  ];
+
+  postPatch = ''
+    sed -i "s@ctypes.util.find_library('libpulse') or 'libpulse.so.0'@'${libpulseaudio}/lib/libpulse.so.0'@" hushboard/_pulsectl.py
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/hushboard/default.nix:46]: Please use substituteInPlace

File: pkgs/applications/audio/hushboard/default.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+, librsvg
+, pycairo
+, pygobject3
+, six
+, xlib
+}:
+
+buildPythonApplication {
+  pname = "hushboard";
+  version = "unstable-2021-03-17";
```
Comment (AndersonTorres) [inline pkgs/applications/audio/hushboard/default.nix:18]:
```suggestion
  version = "0.pre+unstable=2021-03-17";
```

(Shameless plug to future https://github.com/NixOS/rfcs/pull/107; feel free to accept or reject it)

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/hushboard/default.nix [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      --replace "ctypes.util.find_library('libpulse') or 'libpulse.so.0'" "'${libpulseaudio}/lib/libpulse.so.0'"
+  '';
+
+  postInstall = ''
+    # Fix tray icon, see e.g. https://github.com/NixOS/nixpkgs/pull/43421
+    wrapProgram $out/bin/hushboard \
+      --set GDK_PIXBUF_MODULE_FILE "${librsvg.out}/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache"
+
+    mkdir -p $out/share/applications
+    cp snap/gui/hushboard.desktop $out/share/applications
```
Comment (legendofmiracles) [inline pkgs/applications/audio/hushboard/default.nix:56]:
```suggestion
    copyDesktopIcons snap/gui/hushboard.desktop
```
Add `copyDesktopIcons` as a input and to nativeBuildInputs

File: pkgs/applications/audio/hushboard/default.nix [hunk 5]
```diff
@@ -0,0 +1,69 @@
+{ lib
+, buildPythonApplication
+, fetchFromGitHub
+, wrapGAppsHook
```
Comment (AndersonTorres) [inline pkgs/applications/audio/hushboard/default.nix:4]: Put these lists in alphabetical order, except for the three first lines. It helps reading the code.

Do the same on the other lists as well.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Pull request merged**